api/src/main/java/com/github/retrooper/packetevents/util/MojangAPIUtil.java

```diff
@@ -76,6 +77,51 @@ public static List<TextureProperty> requestPlayerTextureProperties(UUID uuid) {
         return null;
     }
 
+    /**
+     * Here you can get a full User Profile with Texture in only one HTTP Request
+     */
+    public static UserProfile createUserProfile(UUID uuid) {
+        String uuidStr = UUIDUtil.toStringWithoutDashes(uuid);
+        try {
+            List<TextureProperty> textureProperties = new ArrayList<>();
+            URL url = new URL("https://sessionserver.mojang.com/session/minecraft/profile/" + uuidStr + "?unsigned=false");
+            HttpURLConnection connection = (HttpURLConnection) url.openConnection();
+            connection.setRequestMethod("GET");
+            //Bad request, this UUID is not valid
+            if (connection.getResponseCode() != 200) {
+                throw new IllegalStateException("Failed to request texture properties with their UUID " + uuidStr + "! Response code: " + connection.getResponseCode());
+            }
+            BufferedReader in = new BufferedReader(
```

Comment:
```suggestion
            try (BufferedReader in = new BufferedReader(new InputStreamReader(connection.getInputStream()))) {
```
Use "try-with" instead of just closing it. And please do it everywhere! (Check if the class implements the `AutoClosable` interface, if so, use try-with :))

Reply:
Okay I just copied from the Get Textures Method above,Yes you can use try-with-resources